js/libs/core.js

```diff
@@ -662,39 +670,73 @@ Fliplet.FormBuilder = (function() {
       var hasOptions = component.props.options && Array.isArray(component.props.options.type());
       var hasSelectAll = component.props.addSelectAll && typeof component.props.addSelectAll.default === 'boolean';
       var isSlider = component.props._componentName.default === 'flSlider';
+      var isMatrix = component.props._componentName.default === 'flMatrix';
 
-      // If options is an array, automatically deal with options
-      if (hasOptions) {
-        component.computed._options = function generateOptions() {
-          return this.options.map(function(option) {
-            if (option.id && option.label && option.id != option.label) {
-              return option.label + ' <' + option.id + '>';
-            }
+      /**
+      * Generate text configurations for radio/checkbox options, separated by new lines
+      * @param {Array} options - A list of options to be mapped
+      * @returns {String} Text options for the configuration interface
+      */
+      var generateOptionsAsText = function(options) {
+        if (!options) {
+          return;
+        }
 
-            return option.label || option.id;
-          }).join('\r\n');
-        };
+        return options.map(function(option) {
+          if (option.id && option.label && option.id !== option.label) {
+            return option.label + ' <' + option.id + '>';
+          }
 
-        component.methods._setOptions = function setOptions(str) {
-          this.options = _.compact(str.split(/\r?\n/).map(function(rawOption) {
-            rawOption = rawOption.trim();
+          return option.label || option.id;
+        }).join('\r\n');
+      };
 
-            var regex = /<.*>$/g;
-            var match = rawOption.match(regex);
-            var option = {};
+      // If options is an array, automatically deal with options
+      if (hasOptions || isMatrix) {
+        if (isMatrix) {
+          component.computed._rowOptions = function generateOptions() {
```

Comment:
You're using the same function name 3 times in this section of the code

```suggestion
          component.computed._rowOptions = function generateRowOptions() {
```